app/src/main/java/com/example/taskyy/ui/screens/ReminderScreen.kt

```diff
@@ -816,3 +828,65 @@ fun SaveTitle(details: String, onEvent: (EditTextEvent) -> Unit) {
     )
 }
 
+@Composable
+fun CheckForNotificationPermissions() {
+    val context = LocalContext.current
+    var hasNotificationPermission by remember {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            mutableStateOf(
+                ContextCompat.checkSelfPermission(
+                    context,
+                    Manifest.permission.POST_NOTIFICATIONS,
+                ) == PackageManager.PERMISSION_GRANTED,
+            )
+        } else {
+            mutableStateOf(true)
+        }
+    }
+
+    val permissionLauncher =
+        rememberLauncherForActivityResult(
+            contract = ActivityResultContracts.RequestPermission(),
+            onResult = { isGranted ->
+                hasNotificationPermission = isGranted
+                // TODO: Handle permission denial later
+            },
+        )
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU && !hasNotificationPermission) {
+        SideEffect {
+            permissionLauncher.launch(Manifest.permission.POST_NOTIFICATIONS)
+        }
+    }
+}
+
+//For some reason this permission check doesn't work
+@Composable
+fun CheckForAlarmPermissions() {
+    val context = LocalContext.current
+    var hasAlarmPermission by remember {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            mutableStateOf(
+                ContextCompat.checkSelfPermission(
+                    context,
+                    Manifest.permission.SCHEDULE_EXACT_ALARM,
+                ) == PackageManager.PERMISSION_GRANTED,
+            )
+        } else {
+            mutableStateOf(true)
+        }
+    }
+
+    val alarmPermissionLauncher =
+        rememberLauncherForActivityResult(
+            contract = ActivityResultContracts.RequestPermission(),
+            onResult = { isGranted ->
+                hasAlarmPermission = isGranted
+                // TODO: Handle permission denial later
+            },
+        )
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU && !hasAlarmPermission) {
+        SideEffect {
+            alarmPermissionLauncher.launch(Manifest.permission.SCHEDULE_EXACT_ALARM)
+        }
+    }
+}
```

Comment:
On Android, there are multiple different protection levels for permissions. They determine how dangerous a permission is and what should be done to grant it. Scheduling exact alarms is not considered as very dangerous that explicitly requires the user's approval, so this requesting is not necessary. That's also why you don't see any dialog for this.
On the other hand, a permission like recording audio is a dangerous one since it could be quickly abused, so the user needs to grant it.

Either way, all permissions an app needs will be listed in the Google Play listing. You can read more about these levels here: https://developer.android.com/guide/topics/manifest/permission-element